{
    "active_lock_reason": null,
    "assignee": {
        "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
        "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
        "followers_url": "https://api.github.com/users/yohaimagen/followers",
        "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
        "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/yohaimagen",
        "id": 12828255,
        "login": "yohaimagen",
        "node_id": "MDQ6VXNlcjEyODI4MjU1",
        "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
        "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
        "repos_url": "https://api.github.com/users/yohaimagen/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/yohaimagen"
    },
    "assignees": [
        {
            "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
            "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
            "followers_url": "https://api.github.com/users/yohaimagen/followers",
            "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
            "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/yohaimagen",
            "id": 12828255,
            "login": "yohaimagen",
            "node_id": "MDQ6VXNlcjEyODI4MjU1",
            "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
            "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
            "repos_url": "https://api.github.com/users/yohaimagen/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/yohaimagen"
        },
        {
            "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
            "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
            "followers_url": "https://api.github.com/users/maxrjones/followers",
            "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
            "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/maxrjones",
            "id": 14077947,
            "login": "maxrjones",
            "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
            "organizations_url": "https://api.github.com/users/maxrjones/orgs",
            "received_events_url": "https://api.github.com/users/maxrjones/received_events",
            "repos_url": "https://api.github.com/users/maxrjones/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/maxrjones"
        }
    ],
    "author_association": "CONTRIBUTOR",
    "body": "`grdimage` throwing a segmentation fault insted of some proper error when mistakenly swapping `xarray` y and x coordination.\r\n \r\n\r\n\r\n**Full code that generated the error**\r\n\r\n```python\r\nX, Y = np.meshgrid(np.arange(10), np.arange(10))\r\nZ = X ** 2 + Y ** 2\r\nda = xr.DataArray(\r\n    data=Z,\r\n    dims=['y', 'x'],\r\n    coords=dict(\r\n        x=(['x'], Y[0, :]),\r\n        y=(['y'], X[:, 0])\r\n    )\r\n)\r\n\r\nfig = pygmt.Figure()\r\nfig.basemap(\r\n    frame='a',\r\n    projection=f\"X10\",\r\n    region=[0, 10, 0, 10]\r\n)\r\nfig.grdimage(da)\r\nfig.show()\r\n```\r\n\r\n**Full error message**\r\n\r\n```\r\nProcess finished with exit code 139 (interrupted by signal 11: SIGSEGV)\r\n\r\n```\r\n\r\n**System information**\r\n\r\nPlease paste the output of `python -c \"import pygmt; pygmt.show_versions()\"`:\r\n\r\n```\r\nPyGMT information:\r\n  version: v0.4.1\r\nSystem information:\r\n  python: 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 15:59:12)  [Clang 11.0.1 ]\r\n  executable: /Users/yohai/anaconda3/envs/faultSlip/bin/python\r\n  machine: Darwin-18.7.0-x86_64-i386-64bit\r\nDependency information:\r\n  numpy: 1.19.1\r\n  pandas: 1.1.1\r\n  xarray: 0.18.2\r\n  netCDF4: 1.5.7\r\n  packaging: 21.0\r\n  ghostscript: 9.50\r\n  gmt: 6.2.0\r\nGMT library information:\r\n  binary dir: /Users/yohai/anaconda3/envs/faultSlip/bin\r\n  cores: 12\r\n  grid layout: rows\r\n  library path: /Users/yohai/anaconda3/envs/faultSlip/lib/libgmt.dylib\r\n  padding: 2\r\n  plugin dir: /Users/yohai/anaconda3/envs/faultSlip/lib/gmt/plugins\r\n  share dir: /Users/yohai/anaconda3/envs/faultSlip/share/gmt\r\n  version: 6.2.0\r\n```\r\n",
    "closed_at": "2021-09-16T18:55:26Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
        "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
        "followers_url": "https://api.github.com/users/maxrjones/followers",
        "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
        "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/maxrjones",
        "id": 14077947,
        "login": "maxrjones",
        "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
        "organizations_url": "https://api.github.com/users/maxrjones/orgs",
        "received_events_url": "https://api.github.com/users/maxrjones/received_events",
        "repos_url": "https://api.github.com/users/maxrjones/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/maxrjones"
    },
    "comments": 9,
    "comments_content": [
        {
            "author_association": "MEMBER",
            "body": "There is something wrong with your x and y coords.\r\n\r\nPrinting the `da` variable gives me:\r\n```\r\n<xarray.DataArray (y: 10, x: 10)>\r\narray([[  0,   1,   4,   9,  16,  25,  36,  49,  64,  81],\r\n       [  1,   2,   5,  10,  17,  26,  37,  50,  65,  82],\r\n       [  4,   5,   8,  13,  20,  29,  40,  53,  68,  85],\r\n       [  9,  10,  13,  18,  25,  34,  45,  58,  73,  90],\r\n       [ 16,  17,  20,  25,  32,  41,  52,  65,  80,  97],\r\n       [ 25,  26,  29,  34,  41,  50,  61,  74,  89, 106],\r\n       [ 36,  37,  40,  45,  52,  61,  72,  85, 100, 117],\r\n       [ 49,  50,  53,  58,  65,  74,  85,  98, 113, 130],\r\n       [ 64,  65,  68,  73,  80,  89, 100, 113, 128, 145],\r\n       [ 81,  82,  85,  90,  97, 106, 117, 130, 145, 162]])\r\nCoordinates:\r\n  * x        (x) int64 0 0 0 0 0 0 0 0 0 0\r\n  * y        (y) int64 0 0 0 0 0 0 0 0 0 0\r\n```\r\nas you can see, x and y coords are all zeros.",
            "created_at": "2021-09-03T13:18:21Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912533525",
            "id": 912533525,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842ZCgV",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912533525/reactions"
            },
            "updated_at": "2021-09-03T13:18:21Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912533525",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "That's true, I didn't notice. It's because of the way I'm building the `DataArray` with the `meshgrid`. Still, I'll consider this a bug because of the segmentation fault, some error should be raise. ",
            "created_at": "2021-09-03T13:45:14Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912551760",
            "id": 912551760,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842ZG9Q",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912551760/reactions"
            },
            "updated_at": "2021-09-03T13:45:26Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912551760",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
                "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
                "followers_url": "https://api.github.com/users/yohaimagen/followers",
                "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
                "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/yohaimagen",
                "id": 12828255,
                "login": "yohaimagen",
                "node_id": "MDQ6VXNlcjEyODI4MjU1",
                "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
                "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
                "repos_url": "https://api.github.com/users/yohaimagen/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/yohaimagen"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> I'll consider this a bug because of the segmentation fault, some error should be raise.\r\n\r\nI agree, but I'm not sure if the error should be done by PyGMT or by GMT.\r\n\r\nPing @meghanrjones ",
            "created_at": "2021-09-03T13:54:09Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912557731",
            "id": 912557731,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842ZIaj",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912557731/reactions"
            },
            "updated_at": "2021-09-03T13:54:09Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912557731",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> > I'll consider this a bug because of the segmentation fault, some error should be raise.\r\n> \r\n> I agree, but I'm not sure if the error should be done by PyGMT or by GMT.\r\n> \r\n> Ping @meghanrjones\r\n\r\nI think ideally it should be caught by both so that PyGMT users have a more readable call stack and other wrappers do not get the same problem. If the same grid is a actual file rather than a remote file, GMT catches the read error properly which suggests it is an API bug.\r\n\r\nWe could pretty easily add a check for this case in the code section below. I cannot find this documented, but I think that GMT expects monotonically increasing, evenly spaced grids even though COARDS compliant grids can be monotonically increasing or decreasing. If this is the case, we could raise `GMTInvalidInput` if `coord_inc <=0`.\r\n\r\nhttps://github.com/GenericMappingTools/pygmt/blob/d01da5ae66ede8b23d1765554a943adc23b1936d/pygmt/clib/conversion.py#L93-L102",
            "created_at": "2021-09-03T16:07:51Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912650883",
            "id": 912650883,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842ZfKD",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912650883/reactions"
            },
            "updated_at": "2021-09-03T16:07:51Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912650883",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
                "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
                "followers_url": "https://api.github.com/users/maxrjones/followers",
                "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
                "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/maxrjones",
                "id": 14077947,
                "login": "maxrjones",
                "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
                "organizations_url": "https://api.github.com/users/maxrjones/orgs",
                "received_events_url": "https://api.github.com/users/maxrjones/received_events",
                "repos_url": "https://api.github.com/users/maxrjones/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/maxrjones"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "I'll add the error.",
            "created_at": "2021-09-04T04:28:32Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912905078",
            "id": 912905078,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842adN2",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912905078/reactions"
            },
            "updated_at": "2021-09-04T04:28:32Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912905078",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
                "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
                "followers_url": "https://api.github.com/users/yohaimagen/followers",
                "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
                "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/yohaimagen",
                "id": 12828255,
                "login": "yohaimagen",
                "node_id": "MDQ6VXNlcjEyODI4MjU1",
                "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
                "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
                "repos_url": "https://api.github.com/users/yohaimagen/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/yohaimagen"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "actually, GMT is handling well monotonically decreasing grid when passing with `DataArray`\r\n\r\nsee example\r\n```python\r\nX, Y = np.meshgrid(np.linspace(0, -10, 10), np.linspace(0, -10, 10))\r\nZ = X ** 2 + Y ** 2\r\nda = xr.DataArray(\r\n    data=Z,\r\n    dims=['y', 'x'],\r\n    coords=dict(\r\n        x=(['x'], X[0, :]),\r\n        y=(['y'], Y[:, 0])\r\n    )\r\n)\r\n\r\nfig = pygmt.Figure()\r\nfig.basemap(\r\n    frame='a',\r\n    projection=f\"X10\",\r\n    region=[-10, 0, -10, 0]\r\n)\r\nfig.grdimage(da)\r\nfig.show()\r\n```\r\nget us the following\r\n\r\n![monotonically_decreasing_grid](https://user-images.githubusercontent.com/12828255/132083097-bb60470c-753e-4f26-955a-8a48a988595e.png)\r\n\r\nso maybe we should allow it and only raise an error when `coord_inc ==0`?\r\n",
            "created_at": "2021-09-04T05:07:01Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-912909104",
            "id": 912909104,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842aeMw",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912909104/reactions"
            },
            "updated_at": "2021-09-04T05:07:01Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/912909104",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
                "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
                "followers_url": "https://api.github.com/users/yohaimagen/followers",
                "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
                "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/yohaimagen",
                "id": 12828255,
                "login": "yohaimagen",
                "node_id": "MDQ6VXNlcjEyODI4MjU1",
                "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
                "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
                "repos_url": "https://api.github.com/users/yohaimagen/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/yohaimagen"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> so maybe we should allow it and only raise an error when coord_inc ==0?\r\n\r\nSounds good.",
            "created_at": "2021-09-06T16:06:45Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-913748389",
            "id": 913748389,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842drGl",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/913748389/reactions"
            },
            "updated_at": "2021-09-06T16:06:45Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/913748389",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
                "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
                "followers_url": "https://api.github.com/users/maxrjones/followers",
                "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
                "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/maxrjones",
                "id": 14077947,
                "login": "maxrjones",
                "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
                "organizations_url": "https://api.github.com/users/maxrjones/orgs",
                "received_events_url": "https://api.github.com/users/maxrjones/received_events",
                "repos_url": "https://api.github.com/users/maxrjones/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/maxrjones"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "I'm reopening the issue because a similar fix should be done in the upstream GMT code.\r\n\r\n@PaulWessel @meghanrjones I think the problem is the API function `GMT_Create_Data` doesn't check if the parameter `wesn` is reasonable (in this case, `wesn=[0, 0, 0, 0]` is passed).",
            "created_at": "2021-09-13T00:05:03Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-917738537",
            "id": 917738537,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842s5Qp",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/917738537/reactions"
            },
            "updated_at": "2021-09-13T00:05:03Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/917738537",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> I'm reopening the issue because a similar fix should be done in the upstream GMT code.\r\n> \r\n> @PaulWessel @meghanrjones I think the problem is the API function `GMT_Create_Data` doesn't check if the parameter `wesn` is reasonable (in this case, `wesn=[0, 0, 0, 0]` is passed).\r\n\r\nThanks for the information. @PaulWessel, I would like to work on fixing this issue. I'll submit a PR in the GMT repository for your feedback and ping if help is needed. We should remember to run gmt.jl tests before merging anything.",
            "created_at": "2021-09-13T21:01:21Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483#issuecomment-918572232",
            "id": 918572232,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
            "node_id": "IC_kwDOBRZfO842wEzI",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/918572232/reactions"
            },
            "updated_at": "2021-09-13T21:01:21Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/918572232",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/14077947?v=4",
                "events_url": "https://api.github.com/users/maxrjones/events{/privacy}",
                "followers_url": "https://api.github.com/users/maxrjones/followers",
                "following_url": "https://api.github.com/users/maxrjones/following{/other_user}",
                "gists_url": "https://api.github.com/users/maxrjones/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/maxrjones",
                "id": 14077947,
                "login": "maxrjones",
                "node_id": "MDQ6VXNlcjE0MDc3OTQ3",
                "organizations_url": "https://api.github.com/users/maxrjones/orgs",
                "received_events_url": "https://api.github.com/users/maxrjones/received_events",
                "repos_url": "https://api.github.com/users/maxrjones/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/maxrjones/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/maxrjones/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/maxrjones"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483/comments",
    "created_at": "2021-09-03T13:08:27Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1483",
    "id": 987755314,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        },
        {
            "color": "e8ff6b",
            "default": false,
            "description": "Bug or missing feature of upstream core GMT",
            "id": 2059525605,
            "name": "upstream",
            "node_id": "MDU6TGFiZWwyMDU5NTI1NjA1",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/upstream"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": "2021-10-31T23:31:11Z",
        "closed_issues": 137,
        "created_at": "2021-03-16T21:12:52Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
            "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
            "followers_url": "https://api.github.com/users/weiji14/followers",
            "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
            "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/weiji14",
            "id": 23487320,
            "login": "weiji14",
            "node_id": "MDQ6VXNlcjIzNDg3MzIw",
            "organizations_url": "https://api.github.com/users/weiji14/orgs",
            "received_events_url": "https://api.github.com/users/weiji14/received_events",
            "repos_url": "https://api.github.com/users/weiji14/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/weiji14"
        },
        "description": "https://github.com/GenericMappingTools/pygmt/issues/1576",
        "due_on": "2021-10-29T07:00:00Z",
        "html_url": "https://github.com/GenericMappingTools/pygmt/milestone/8",
        "id": 6553956,
        "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/8/labels",
        "node_id": "MDk6TWlsZXN0b25lNjU1Mzk1Ng==",
        "number": 8,
        "open_issues": 0,
        "state": "closed",
        "title": "0.5.0",
        "updated_at": "2022-09-22T11:49:00Z",
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/8"
    },
    "node_id": "MDU6SXNzdWU5ODc3NTUzMTQ=",
    "number": 1483,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483/timeline",
    "title": "Segmentation fault if a grid has zero increments",
    "updated_at": "2021-09-16T18:55:26Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1483",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/12828255?v=4",
        "events_url": "https://api.github.com/users/yohaimagen/events{/privacy}",
        "followers_url": "https://api.github.com/users/yohaimagen/followers",
        "following_url": "https://api.github.com/users/yohaimagen/following{/other_user}",
        "gists_url": "https://api.github.com/users/yohaimagen/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/yohaimagen",
        "id": 12828255,
        "login": "yohaimagen",
        "node_id": "MDQ6VXNlcjEyODI4MjU1",
        "organizations_url": "https://api.github.com/users/yohaimagen/orgs",
        "received_events_url": "https://api.github.com/users/yohaimagen/received_events",
        "repos_url": "https://api.github.com/users/yohaimagen/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/yohaimagen/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/yohaimagen/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/yohaimagen"
    }
}